iroh-net/src/magicsock/node_map.rs

```diff
@@ -85,11 +86,9 @@ enum NodeStateKey {
 }
 
 /// Source for a new node.
-///
-/// This is used for debugging purposes.
-#[derive(strum::Display, Debug)]
+#[derive(Serialize, Deserialize, strum::Display, Debug, Clone, Eq, PartialEq, Hash)]
 #[strum(serialize_all = "kebab-case")]
-pub(crate) enum Source {
+pub enum Source {
```

Comment:
Now that this is pub it definitely needs more docs.

* Is this a source for a node or a source for a network path to a node?  (Genuine question, I'm not actually sure and would have to carefully re-read this whole PR!  Maybe it's both!)
* What is "new" about this node? 
* What is considered a source?  Sounds silly but a sentence or two saying that is how we know about the existence of a node is appropriate to me.  Most docs should try and stand on themselves a bit without having to understand the entire system.
* Where might I encounter this?  Seems like a reference to (custom discovery mechanisms as well as `Endpoint::add_node_addr_with_source` would not be amiss.

I also chased down what `Saved` means as this was surprising to me - iroh-net does not interact with the filesystem.  It turns out it is used for any `NodeAddr` added using `iroh_net::endpoint::Builder::known_nodes`.  I don't know what to suggest here, but "saved" seems hardly accurate.  Maybe `App` would be more suitable, but that leads to...

Let's chase down why we have `App` and `NamedApp`!  `App` seems to come directly from `Endpoint::add_node_addr`, that's a pretty prominent API.  `NamedApp` comes from `Endpoint::add_node_addr_with_source`.  I think this relation should be described in the items here because otherwise someone looking at the sources here is just as confused about it as I am.

I despair at myself for caring about these things.  The rabbit holes won't end.

Reply:
`Saved` is a remnant from when persistence was inside iroh-net, and should def be removed. About `App`, I see your concerns but I'm not sure how to best solve it. This is akin to `Unknown`. except that it's not entirely unknown, it was manually added via the api, in contrast to automatically added via discovery, for example. Maybe based on this we can come up with a better name

Reply:
I think it's fine to keep `App`, but acknowledge in it's docs that `NamedApp` also exists and vice-versa.